tests/async/test_browsercontext_cookies.py

```diff
@@ -0,0 +1,189 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import pytest
+
+
+async def test_should_return_no_cookies_in_pristine_browser_context(context):
+    assert await context.cookies() == []
+
+
+async def test_should_get_a_cookie(context, page, server):
+    await page.goto(server.EMPTY_PAGE)
+    document_cookie = await page.evaluate(
+        """() => {
+    document.cookie = 'username=John Doe';
+    return document.cookie;
+  }"""
+    )
+    assert document_cookie == "username=John Doe"
+    assert await context.cookies() == [
+        {
+            "name": "username",
+            "value": "John Doe",
+            "domain": "localhost",
+            "path": "/",
+            "expires": -1,
+            "httpOnly": False,
+            "secure": False,
+            "sameSite": "None",
+        }
+    ]
+
+
+async def test_should_get_a_non_session_cookie(context, page, server):
+    await page.goto(server.EMPTY_PAGE)
+    # @see https://en.wikipedia.org/wiki/Year_2038_problem
+    date = int(datetime.datetime(2038, 1, 1).timestamp() * 1000)
+    document_cookie = await page.evaluate(
+        """timestamp => {
+    const date = new Date(timestamp);
+    document.cookie = `username=John Doe;expires=${date.toUTCString()}`;
+    return document.cookie;
+  }""",
+        date,
+    )
+    assert document_cookie == "username=John Doe"
+    assert await context.cookies() == [
+        {
+            "name": "username",
+            "value": "John Doe",
+            "domain": "localhost",
+            "path": "/",
+            "expires": date / 1000,
+            "httpOnly": False,
+            "secure": False,
+            "sameSite": "None",
+        }
+    ]
+
+
+async def test_should_properly_report_httponly_cookie(context, page, server):
+    server.set_route(
+        "/empty.html",
+        lambda r: (
+            r.setHeader("Set-Cookie", "name=value;HttpOnly; Path=/"),
+            r.finish(),
+        ),
+    )
+
+    await page.goto(server.EMPTY_PAGE)
+    cookies = await context.cookies()
+    assert len(cookies) == 1
+    assert cookies[0]["httpOnly"] == True
+
+
+# FIXME: Pytest is going to skip too much here
+# it.fail(WEBKIT && WIN)
+@pytest.mark.skip_platform("win")
+@pytest.mark.skip_browser("webkit")
+async def test_should_properly_report_strict_samesite_cookie(context, page, server):
+    server.set_route(
+        "/empty.html",
+        lambda r: (
+            r.setHeader("Set-Cookie", "name=value;SameSite=Strict"),
+            r.finish(),
+        ),
+    )
+    await page.goto(server.EMPTY_PAGE)
+    cookies = await context.cookies()
+    assert len(cookies) == 1
+    assert cookies[0]["sameSite"] == "Strict"
+
+
+# FIXME: Pytest is going to skip too much here
+# it.fail(WEBKIT && WIN)
```

Comment:
@mxschmitt What's the proper way for me to annotate this in pytest? As is, I believe it will skip `webkit || win`, but I want `webkit && win` to preserve the JS port.

Reply:
I did it like that: https://github.com/microsoft/playwright-python/blob/8c5cb546280ec0cdacbc378f29092e034502449e/tests/async/test_headful.py#L36

Reply:
Thanks!